test/functionalTest/cases/1887_hash_in_url_gives_strange_results/hash_in_url_gives_strange_results.test

```diff
@@ -0,0 +1,120 @@
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Hash sign in URL gives strange results
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. GET /v2/entities/E1_#/attrs/A1/value
+# 02. GET /v2/entities/E1/attrs/A1_#/value
+# 03. GET /v2/entities/E1_#/attrs/A1/value, Accept text/plain
+# 04. GET /v2/entities/E1/attrs/A1_#/value, Accept text/plain
+#
+
+echo '01. GET /v2/entities/E1_#/attrs/A1/value'
+echo '========================================'
+orionCurl --url '/v2/entities/E1_#/attrs/A1/value'
+echo
+echo
+
+
+echo '02. GET /v2/entities/E1/attrs/A1_#/value'
+echo '========================================'
+orionCurl --url '/v2/entities/E1/attrs/A1_#/value'
+echo
+echo
+
+
+echo '03. GET /v2/entities/E1_#/attrs/A1/value, Accept text/plain'
+echo '==========================================================='
+orionCurl --url '/v2/entities/E1_#/attrs/A1/value' --out text/plain
+echo
+echo
+
+
+echo '04. GET /v2/entities/E1/attrs/A1_#/value, Accept text/plain'
+echo '==========================================================='
+orionCurl --url '/v2/entities/E1/attrs/A1_#/value' --out text/plain
+echo
+echo
+
+
+--REGEXPECT--
+01. GET /v2/entities/E1_#/attrs/A1/value
+========================================
+HTTP/1.1 400 Bad Request
+Content-Length: 63
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "invalid character in URI",
+    "error": "BadRequest"
+}
+
+
+02. GET /v2/entities/E1/attrs/A1_#/value
+========================================
+HTTP/1.1 400 Bad Request
+Content-Length: 63
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "invalid character in URI",
+    "error": "BadRequest"
+}
+
+
+03. GET /v2/entities/E1_#/attrs/A1/value, Accept text/plain
+===========================================================
+HTTP/1.1 400 Bad Request
+Content-Length: 63
+Content-Type: text/plain
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{"error":"BadRequest","description":"invalid character in URI"}
```

Comment:
When orionCurl receives a response for a request sent with Accept: text/plain, orionCurl assumes the response is normal text and does **not** try to beautify the response with any JSON beautifier.

This is the reason why the response is shown 'raw' here and in the next step.

[ About getting JSON back when asking for plain text ... see the description of the PR ]

NTC, just informative